app/controllers/photos_controller.rb

```diff
@@ -0,0 +1,24 @@
+class PhotosController < ApplicationController
+  before_filter :authorize
+
+  def update
+    @photo = Pin.find(params[:pin_id]).photos.find(params[:id])
```

Comment:
should use Photo.find(params[:id]) only to reduce the number of query to database